scripts/pylib/twister/twisterlib/twister_main.py

```diff
@@ -201,12 +205,12 @@ def main(options):
 
     coverage_completed = True
     if options.coverage:
-        if not options.build_only:
+        if not options.build_only or not options.report_summary:
             coverage_completed = run_coverage(tplan, options)
         else:
-            logger.info("Skipping coverage report generation due to --build-only.")
+            logger.info("Skipping coverage report generation due to --build-only or --report-summary.")
```

Comment:
this check can be done early on command line parsing with error code returned.

Reply:
outdated